src/nidigital/system_tests/test_system_nidigital.py

```diff
@@ -961,6 +961,121 @@ def test_configure_pattern_burst_sites(multi_instrument_session):
+        write_waveform_data=True)
+
+    # burst the waveform to confirm that configuration and loading went okay
+    pass_fail = multi_instrument_session.burst_pattern(start_label='new_pattern')
+    assert pass_fail == {0: True, 1: True}
+
+
+def test_write_source_waveform_data_from_file_tdms(multi_instrument_session):
+    test_name = 'test_source_waveform_parallel_broadcast'
+    configure_session(multi_instrument_session, test_name)
+    multi_instrument_session.load_pattern(get_test_file_path(test_name, 'pattern.digipat'))
+
+    multi_instrument_session.create_source_waveform_from_file_tdms(
+        waveform_name='src_wfm',
+        waveform_file_path=get_test_file_path(test_name, 'source_waveform.tdms'),
+        write_waveform_data=False)
+    try:  # confirm that the waveform is not yet loaded
+        multi_instrument_session.burst_pattern(start_label='new_pattern')
+        assert False
+    except nidigital.Error as e:
+        assert e.code == -1074118614
+        assert e.description.find('The source waveform(s) used in the pattern(s) to be burst have not been written to source memory.'
+                                  ' Ensure that you write source waveforms with niDigital Write Source Waveform.') != -1
+
+    multi_instrument_session.write_source_waveform_data_from_file_tdms(
+        waveform_name='src_wfm',
+        waveform_file_path=get_test_file_path(test_name, 'source_waveform.tdms'))
+
+    # burst the waveform to confirm that configuration and loading went okay
+    pass_fail = multi_instrument_session.burst_pattern(start_label='new_pattern')
+    assert pass_fail == {0: True, 1: True}
+
+
+def test_create_capture_waveform_serial(multi_instrument_session):
+    test_name = 'test_fetch_capture_waveform'
+    configure_session(multi_instrument_session, test_name)
+    multi_instrument_session.load_pattern(get_test_file_path(test_name, 'pattern.digipat'))
+    num_samples = 256
+
+    multi_instrument_session.pins['LO0'].create_capture_waveform_serial(
+        waveform_name='capt_wfm',
+        sample_width=2,
+        bit_order=nidigital.BitOrder.LSB)
+
+    # The pattern references a wfm 'src_wfm', so we have to load it before we can burst
+    multi_instrument_session.pins['LO0'].create_source_waveform_serial(
+        waveform_name='src_wfm',
+        data_mapping=nidigital.SourceDataMapping.BROADCAST,
+        sample_width=2,
+        bit_order=nidigital.BitOrder.LSB)
+    multi_instrument_session.write_source_waveform_broadcast(
+        waveform_name='src_wfm',
+        waveform_data=[0])
+    multi_instrument_session.burst_pattern(start_label='new_pattern')
+
+    # Fetch to confirm that configuration went okay
+    multi_instrument_session.sites[1, 0].fetch_capture_waveform(
+        waveform_name='capt_wfm',
+        samples_to_read=num_samples)
+
+
+def test_create_capture_waveform_from_file_digicapture(multi_instrument_session):
```

Comment:
Almost all comments for `test_create_capture_waveform_serial` apply for this test too.

Reply:
Updated.
I decided to point this test at the project used by `test_create_capture_waveform_serial` and load the source from the .tdms file.